### DISCUSSION TITLE
Possibility to extract anc data?

### DISCUSSION THREAD

**Discussion by acastr** [bbc/bmx#10] [Q&A]
I'm trying to extract a specific data track inside an mxf. I know the data is there if i use mxf2raw; (example output below)

I know how to decode the raw hex data inside the vanc but I haven't been able to extract the specific track to text (or any other format)

I'm trying to build a cli app that can extract all SCTE data from a recorded mxf file, but I'm failing already at being able to extract the data track.

Could you please point me into the right direction? (although it might not seem that way, I already tried the various exe's supplied but it seems I'm not finding the correct option flags)

Or any other approach if I'm not approaching this in the proper way.
Kind regards,
Anthony
e.g.
 (created='2023-04-03 10:45:25.000')
Application
  name            : mxf2raw
  lib_name        : bmx
  lib_version     : 0.1.3
  build_timestamp : 2021-07-07 15:36:49.000
  scm_version     : snap1708-108-ga304cf7
Files: (1)
  File #0:
    file_uri        : file:///D:/Applicatie_cache/castrea/Downloads/bmx-win32-exe-snapshot-20220831/SCTE_5.mxf
    mxf_version     : 1.3
    op_label        : OP1A (external_essence='false', non_stream='false', multi_track='true', label='urn:smpte:ul:060e2b34.04010101.0d010201.01010900')
    frame_wrapped   : true
    Identifications: (1)
      Identification #0:
        generation_uid  : urn:uuid:d28afa5e-92bd-1d1e-b4a6-00d0281f79b5
        company_name    : Omneon Inc.
        product_name    : Omneon Media Subsystem
        product_version : 9.5.0.0.1
        version_string  : 9.5.0.0.release
        product_uid     : urn:uuid:00000000-0000-1000-8000-050e0b010602
        modified_date   : 2023-03-08 09:19:47.468
        platform        : Omneon Media Api (mqx)    
    MaterialPackages: (1)
      MaterialPackage #0:
        package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.13001729.008ffa5e.92bd1d1e.bf7700d0.281f79b5
        creation_date   : 2023-03-08 09:19:47.468
        modified_date   : 2023-03-08 09:19:47.468
    FileSourcePackages: (1)
      FileSourcePackage #0:
        package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.1300a048.be8ffa5e.92bd1d1e.976c00d0.281f79b5
        creation_date   : 2023-03-08 09:19:47.468
        modified_date   : 2023-03-08 09:19:47.468
Clip
  edit_rate       : 25/1
  duration        : 00:03:01:01 (count='4526')
  StartTimecodes:
    material        : 10:19:57:23
    file_source     : 10:19:57:23
  Tracks: (3)
    Track #0:
      essence_kind    : Picture
      essence_type    : AVCI_100_1080i
      ec_label        : urn:smpte:ul:060e2b34.0401010a.0d010301.02106001
      edit_rate       : 25/1
      duration        : 00:03:01:01 (count='4526')
      Packages: (1)
        Package #0:
          Material:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.13001729.008ffa5e.92bd1d1e.bf7700d0.281f79b5
            track_id        : 2
            track_number    : 0
          FileSource:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.1300a048.be8ffa5e.92bd1d1e.976c00d0.281f79b5
            track_id        : 2
            track_number    : 0x15010500
            file_uri        : file:///D:/Applicatie_cache/castrea/Downloads/bmx-win32-exe-snapshot-20220831/SCTE_5.mxf
      PictureDescriptor:
        coding_label    : urn:smpte:ul:060e2b34.0401010a.04010202.01323102
        signal_standard : None (value='0')
        frame_layout    : Separate_Fields (value='1')
        stored_width    : 1920
        stored_height   : 1088
        display_width   : 1920
        display_height  : 1080
        aspect_ratio    : 16/9
        afd             : 8
        avci_header     : true
        CDCIDescriptor:
          component_depth : 10
          horiz_subsamp   : 2
          vert_subsamp    : 1
          color_siting    : Unknown (value='255')
    Track #1:
      essence_kind    : Sound
      essence_type    : WAVE_PCM
      ec_label        : urn:smpte:ul:060e2b34.04010101.0d010301.02060100
      edit_rate       : 25/1
      duration        : 00:03:01:01 (count='4526')
      Packages: (1)
        Package #0:
          Material:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.13001729.008ffa5e.92bd1d1e.bf7700d0.281f79b5
            track_id        : 3
            track_number    : 0
          FileSource:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.1300a048.be8ffa5e.92bd1d1e.976c00d0.281f79b5
            track_id        : 3
            track_number    : 0x16010100
            file_uri        : file:///D:/Applicatie_cache/castrea/Downloads/bmx-win32-exe-snapshot-20220831/SCTE_5.mxf
      SoundDescriptor:
        sampling_rate        : 48000/1
        bits_per_sample      : 24
        block_align          : 24
        channel_count        : 8
        locked               : true
    Track #2:
      essence_kind    : Data
      essence_type    : ANC_Data
      ec_label        : urn:smpte:ul:060e2b34.04010109.0d010301.020e0000
      edit_rate       : 25/1
      duration        : 00:03:01:01 (count='4526')
      Packages: (1)
        Package #0:
          Material:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.13001729.008ffa5e.92bd1d1e.bf7700d0.281f79b5
            track_id        : 4
            track_number    : 0
          FileSource:
            package_uid     : urn:smpte:umid:060a2b34.01010105.01010d23.1300a048.be8ffa5e.92bd1d1e.976c00d0.281f79b5
            track_id        : 4
            track_number    : 0x17010201
            file_uri        : file:///D:/Applicatie_cache/castrea/Downloads/bmx-win32-exe-snapshot-20220831/SCTE_5.mxf
      DataDescriptor:
        ANCDescriptor:
          Manifest: (4)
            Element #0:
              line_number     : 11
              wrapping_type   : VANC_Frame (value='1')
              sample_coding   : ANC_8_Bit_Luma (value='4')
              DIDType2:
                did             : 0x41
                sdid            : 0x05
                description     : [S2016-3] AFD and Bar Data
            Element #1:
              line_number     : 13
              wrapping_type   : VANC_Frame (value='1')
              sample_coding   : ANC_8_Bit_Luma (value='4')
              DIDType2:
                did             : 0x41
                sdid            : 0x08
                description     : [S2031] DVB/SCTE VBI data
            Element #2:
              line_number     : 574
              wrapping_type   : VANC_Frame (value='1')
              sample_coding   : ANC_8_Bit_Luma (value='4')
              DIDType2:
                did             : 0x41
                sdid            : 0x05
                description     : [S2016-3] AFD and Bar Data
            Element #3:
              line_number     : 576
              wrapping_type   : VANC_Frame (value='1')
              sample_coding   : ANC_8_Bit_Luma (value='4')
              DIDType2:
                did             : 0x41
                sdid            : 0x08
                description     : [S2031] DVB/SCTE VBI data

**Comment by philipnbbc** [comment]
I can't try it myself atm, but the -p option can be used to extract the raw binary data for all tracks. It'll probably include the ST436 header for the data tracks which is shown in the mxf2raw text dump. The --disable-* options can be used to not extract from tracks you don't want.
